File: app/src/main/java/com/xwm/androidproject/util/ViewUtil.java
package com.xwm.androidproject.util;

import android.util.Log;
import android.view.View;

/**
 * View相关通用逻辑类
 */
public class ViewUtil {
    private static final String TAG = "ViewUtil";

    private static View.OnTouchListener onTouchListener = (view, event) -> {
        view.performClick();
        return true;
    };

    public static void show(View view) {
        if (view != null) {
            view.setVisibility(View.VISIBLE);
        }
    }

    public static void hide(View view) {
        if (view != null) {
            view.setVisibility(View.GONE);
        }
    }

    public static void invisible(View view) {
        if (view != null) {
            view.setVisibility(View.INVISIBLE);
        }
    }

    public static boolean isVisible(View view) {
        if (view == null) {
            return false;
        }
        return view.getVisibility() == View.VISIBLE;
    }

    public static void toggle(View view) {
        if (view != null) {
            view.setVisibility(view.isShown() ? View.GONE : View.VISIBLE);
        }
    }

    public static void showIf(View view, boolean condition) {
        if (view != null) {
            view.setVisibility(condition ? View.VISIBLE : View.GONE);
        }
    }

    /**
     * 消费View的Touch事件，防止touch渗透
     *
     * @param view
     */
    public static void shieldTouchEvent(View view) {
        if (view != null) {
            view.setOnTouchListener(onTouchListener);
        }
    }

    /**
     * 偏移
     *
     * @param view
     * @param xOffset
     * @param yOffset
     */
    public static void offset(View view, int xOffset, int yOffset) {
        int left = view.getLeft() + xOffset;
        int top = view.getTop() + yOffset;
        int width = view.getWidth();
        int height = view.getHeight();
        int right = left + width;
        int bottom = top + height;
        view.layout(left, top, right, bottom);
        Log.d(TAG, "move margin left : " + left + " top : " + top + " view : " + view);
//        view.setTag(R.id.center, null);
    }
}

File: base/src/main/java/com/xwm/base/util/ViewUtil.kt
package com.xwm.base.util

import android.util.Log
import android.view.View
import android.view.ViewGroup

/**
 * View相关通用逻辑类
 */
object ViewUtil {
    private const val TAG = "ViewUtil"

    private val onTouchListener = View.OnTouchListener { _, _ ->
        //        view.performClick()
        true
    }

    fun show(view: View) {
        view.visibility = View.VISIBLE
    }

    fun hide(view: View) {
        view.visibility = View.GONE
    }

    fun invisible(view: View) {
        view.visibility = View.INVISIBLE
    }

    fun isVisible(view: View): Boolean {
        return view.visibility == View.VISIBLE
    }

    fun toggle(view: View) {
        view.visibility = if (view.isShown) View.GONE else View.VISIBLE
    }

    fun showIf(view: View, condition: Boolean) {
        view.visibility = if (condition) View.VISIBLE else View.GONE
    }

    /**
     * 消费View的Touch事件，防止touch渗透
     *
     * @param view
     */
    fun shieldTouchEvent(view: View) {
        view.setOnTouchListener(onTouchListener)
    }

    fun removeFromParent(view: View?) {
        view?.let {
            val parent = view.parent
            if (parent is ViewGroup) {
                parent.removeView(view)
            }
        }
    }

    /**
     * 偏移
     *
     * @param view
     * @param xOffset
     * @param yOffset
     */
    fun offset(view: View, xOffset: Int, yOffset: Int) {
        val left = view.left + xOffset
        val top = view.top + yOffset
        val width = view.width
        val height = view.height
        val right = left + width
        val bottom = top + height
        view.layout(left, top, right, bottom)
        Log.d(TAG, "move margin left : $left top : $top view : $view")
        //        view.setTag(R.id.center, null);
    }
}
